backend/routes/userRoutes.js

```diff
@@ -105,4 +105,142 @@ router.patch('/location', async(req, res) => {
+    eventDate = new Date(parseInt(e.date.slice(0, 2)) + '/' + parseInt(e.date.slice(3, 5)) + '/' + parseInt(e.date.slice(6, 10)))
+    dateDifference =  ( ((eventDate.getTime() - today.getTime()) / 1000) / 604800 ) 
+
+    if (dateDifference !=0 ){
+      dateScore = 1 / dateDifference
+    } else{
+      dateScore = 1
+    }
+    
+    // ------------------------------------------------------------------------------------------------------------------------
+    // Category Score -----------------------------------------------------------------------------------------------------------
+    countOfApperances = 0
+    for (let i = 0; i < categoriesEventsAttending.length; i++) {
+      if (categoriesEventsAttending[i] == e.category){
+        countOfApperances = countOfApperances + 1
+      }
+    }
+    if (categoriesEventsAttending.length != 0) {
+      categoryScore = (countOfApperances)/(categoriesEventsAttending.length)
+    } else {
+      categoryScore = 0
+    }
+    // ------------------------------------------------------------------------------------------------------------------------
+    // Distance Score -----------------------------------------------------------------------------------------------------------
+    distanceEventToUser = ( ((e.lat - user.lat)**2 + (e.long - user.long)**2)**.5 )
+
+    if ((meanDistanceTraveledUser - distanceEventToUser) > 0){
+      distanceScore = 1 / (meanDistanceTraveledUser - distanceEventToUser)
+    } else if ((meanDistanceTraveledUser - distanceEventToUser) < 0) {
+      distanceScore = 1 / (distanceEventToUser - meanDistanceTraveledUser)
+    }
+    else {
+      distanceScore = 1
+    }
+    // -------------------------------------------------------------------------------------------------------------------------
+    // Total Score -----------------------------------------------------------------------------------------------------------
+    totalScore = dateScore + categoryScore + distanceScore
+    // -------------------------------------------------------------------------------------------------------------------------
+
+    idRecommendedEvents.push(e.id)
+    scoreRecommendedEvents.push(totalScore)
+  })
+  // *****************************************************************************************************************************
+  // Sort events by score --------------------------------------------------------------
+  var list = [];
+  for (var j = 0; j < idRecommendedEvents.length; j++) {
+      list.push({'id': idRecommendedEvents[j], 'score': scoreRecommendedEvents[j]});
+  }
+
+  list.sort(function(a, b) {
+      return ((a.score < b.score) ? 1 : ((a.score == b.score) ? 0 : -1));
+  });
+
+  for (var k = 0; k < list.length; k++) {
+      if (k < 10){
+        idRecommendedEvents[k] = list[k].id;
+      }
+  }
+  // ------------------------------------------------------------------------------------
```

Comment:
I recommend turning this into a separate function

Reply:
Done.
Created recommendTenEvents function.
Tested correctness by logging the different metrics in different times